akka-stream/src/main/boilerplate/akka/stream/scaladsl/UnzipWithApply.scala.template

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (C) 2014-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import akka.stream._
+import akka.stream.impl.GenJunctions._
+import akka.stream.impl.StreamLayout
+
+trait UnzipWithApply {
+
+  [2..20#/**
+   * Create a new `UnzipWith` specialized for 1 outputs.
+   *
+   * @param unzipper unzipping-function from the input value to 1 output values
+   */
+  def apply[O, [#A1#]](unzipper: O ⇒ ([#A1#])): UnzipWith1[O, [#A1#]] = {
+    val shape = new FanOutShape1[O, [#A1#]]("UnzipWith1")
+    new UnzipWith1(shape, new UnzipWith1Module(shape, unzipper, OperationAttributes.name("UnzipWith1")))
+    }
+    #
+
+  ]
```

Comment:
I wonder if we could fix the "has same type after erasure" warnings on these apply methods hm...


```
[warn] /mnt/akka/jenkinsakka/localhome-a0/workspace/pr-validator-per-commit-jenkins/akka-stream/target/java/akka/stream/scaladsl/UnzipWithApply.java:114: error: name clash: <O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>apply(Function1<O#1,Tuple17<A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>>) and <O#2,A1#2,A2#2>apply(Function1<O#2,Tuple2<A1#2,A2#2>>) have the same erasure
[warn]   public <O extends java.lang.Object, A1 extends java.lang.Object, A2 extends java.lang.Object, A3 extends java.lang.Object, A4 extends java.lang.Object, A5 extends java.lang.Object, A6 extends java.lang.Object, A7 extends java.lang.Object, A8 extends java.lang.Object, A9 extends java.lang.Object, A10 extends java.lang.Object, A11 extends java.lang.Object, A12 extends java.lang.Object, A13 extends java.lang.Object, A14 extends java.lang.Object, A15 extends java.lang.Object, A16 extends java.lang.Object, A17 extends java.lang.Object> akka.stream.scaladsl.UnzipWith17<O, A1, A2, A3, A4, A5, A6, A7, A8, A9, A10, A11, A12, A13, A14, A15, A16, A17> apply (scala.Function1<O, scala.Tuple17<A1, A2, A3, A4, A5, A6, A7, A8, A9, A10, A11, A12, A13, A14, A15, A16, A17>> unzipper) ;
[warn]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            ^
[warn]   where O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17,O#2,A1#2,A2#2 are type-variables:
[warn]     O#1 extends Object declared in method <O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>apply(Function1<O#1,Tuple17<A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>>)
[warn]     A1#1 extends Object declared in method <O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>apply(Function1<O#1,Tuple17<A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>>)
[warn]     A2#1 extends Object declared in method <O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>apply(Function1<O#1,Tuple17<A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>>)
[warn]     A3 extends Object declared in method <O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>apply(Function1<O#1,Tuple17<A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>>)
[warn]     A4 extends Object declared in method <O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>apply(Function1<O#1,Tuple17<A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>>)
[warn]     A5 extends Object declared in method <O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>apply(Function1<O#1,Tuple17<A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>>)
[warn]     A6 extends Object declared in method <O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>apply(Function1<O#1,Tuple17<A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>>)
[warn]     A7 extends Object declared in method <O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>apply(Function1<O#1,Tuple17<A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>>)
[warn]     A8 extends Object declared in method <O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>apply(Function1<O#1,Tuple17<A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>>)
[warn]     A9 extends Object declared in method <O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>apply(Function1<O#1,Tuple17<A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>>)
[warn]     A10 extends Object declared in method <O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>apply(Function1<O#1,Tuple17<A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>>)
[warn]     A11 extends Object declared in method <O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>apply(Function1<O#1,Tuple17<A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>>)
[warn]     A12 extends Object declared in method <O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>apply(Function1<O#1,Tuple17<A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>>)
[warn]     A13 extends Object declared in method <O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>apply(Function1<O#1,Tuple17<A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>>)
[warn]     A14 extends Object declared in method <O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>apply(Function1<O#1,Tuple17<A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>>)
[warn]     A15 extends Object declared in method <O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>apply(Function1<O#1,Tuple17<A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>>)
[warn]     A16 extends Object declared in method <O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>apply(Function1<O#1,Tuple17<A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>>)
[warn]     A17 extends Object declared in method <O#1,A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>apply(Function1<O#1,Tuple17<A1#1,A2#1,A3,A4,A5,A6,A7,A8,A9,A10,A11,A12,A13,A14,A15,A16,A17>>)
[warn]     O#2 extends Object declared in method <O#2,A1#2,A2#2>apply(Function1<O#2,Tuple2<A1#2,A2#2>>)
[warn]     A1#2 extends Object declared in method <O#2,A1#2,A2#2>apply(Function1<O#2,Tuple2<A1#2,A2#2>>)
[warn]     A2#2 extends Object declared in method <O#2,A1#2,A2#2>apply(Function1<O#2,Tuple2<A1#2,A2#2>>)
```